src/account/operations/output_consolidation.rs

```diff
@@ -22,36 +22,65 @@ const MIN_OUTPUT_SIZE_IN_ESSENCE: usize = 46;
 
 #[cfg(feature = "ledger_nano")]
 use crate::account::constants::DEFAULT_LEDGER_OUTPUT_CONSOLIDATION_THRESHOLD;
+use crate::Result;
 use crate::account::{
     constants::DEFAULT_OUTPUT_CONSOLIDATION_THRESHOLD, handle::AccountHandle,
-    operations::output_claiming::get_new_native_token_count, types::Transaction, TransactionOptions,
+    operations::{
+        helpers::time::can_output_be_unlocked_now,
+        output_claiming::get_new_native_token_count
+    },
+    types::{OutputData, Transaction},
+    AddressWithUnspentOutputs,
+    TransactionOptions,
 };
 
 impl AccountHandle {
+    fn should_consolidate_output(self: &AccountHandle, output_data: &OutputData, current_time: u32, account_addresses: &[AddressWithUnspentOutputs]) -> Result<bool> {
+        Ok(if let Output::Basic(basic_output) = &output_data.output {
+            let unlock_conditions = basic_output.unlock_conditions();
+
+            if unlock_conditions.is_time_locked(current_time) {
+                // We cannot consolidate timelocked outputs
+                return Ok(false);
+            }
+
+            if unlock_conditions.storage_deposit_return().is_none() && unlock_conditions.expiration().is_none() {
```

Comment:
My fault, but should be this :trollface: 
```suggestion
            if unlock_conditions.storage_deposit_return().is_some() && unlock_conditions.expiration().is_none() {
```